dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/URAPIs_ADLDAP_SSLTest.java

```diff
@@ -42,54 +49,189 @@
-        server.addInstalledAppForValidation("userRegistry");
-        server.startServer(c.getName() + ".log");
+        libertyServer.copyFileToLibertyInstallRoot("lib/features", "internalfeatures/securitylibertyinternals-1.0.mf");
+        libertyServer.addInstalledAppForValidation("userRegistry");
+        libertyServer.startServer(c.getName() + ".log");
 
-        //Make sure the application has come up before proceeding
+        /*
+         * Make sure the application has come up before proceeding
+         */
         assertNotNull("Application userRegistry does not appear to have started.",
-                      server.waitForStringInLog("CWWKZ0001I:.*userRegistry"));
+                      libertyServer.waitForStringInLog("CWWKZ0001I:.*userRegistry"));
         assertNotNull("Security service did not report it was ready",
-                      server.waitForStringInLog("CWWKS0008I"));
+                      libertyServer.waitForStringInLog("CWWKS0008I"));
         assertNotNull("Server did not came up",
-                      server.waitForStringInLog("CWWKF0011I"));
+                      libertyServer.waitForStringInLog("CWWKF0011I"));
 
         Log.info(c, "setUp", "Creating servlet connection the server");
-        servlet = new UserRegistryServletConnection(server.getHostname(), server.getHttpDefaultPort());
+        servlet = new UserRegistryServletConnection(libertyServer.getHostname(), libertyServer.getHttpDefaultPort());
 
         if (servlet.getRealm() == null) {
             Thread.sleep(5000);
             servlet.getRealm();
         }
+
+        initialConfiguration = libertyServer.getServerConfiguration();
     }
 
-    @AfterClass
-    public static void tearDown() throws Exception {
-        Log.info(c, "tearDown", "Stopping the server...");
-        try {
-            server.stopServer("CWIML4529E", "CWIML4537E", "CWPKI0041W");
-        } finally {
-            server.deleteFileFromLibertyInstallRoot("lib/features/internalfeatures/securitylibertyinternals-1.0.mf");
-        }
+    /**
+     * Configure the embedded LDAP server.
+     *
+     * @throws Exception If the server failed to start for some reason.
+     */
+    private static void setupLdapServer() throws Exception {
+        String serverPath = libertyServer.getServerRoot();
+        KEYSTORE_PATH = serverPath + "/keystore.jks";
+        LDIF_PATH = serverPath + "/AD.ldif";
+        Log.info(c, "setupLdapServer", "LDIF Path is " + LDIF_PATH);
+        ds = new InMemoryADLDAPServer(BASE_DN);
+        int entriesAdded = ds.importFromLDIF(true, LDIF_PATH);
+        Log.info(c, "setupLdapServer", "Adding " + entriesAdded + " changes to LDAP Server");
+
+    }
+
+    /**
+     * Configure the embedded LDAP server.
+     *
+     * @throws Exception If the server failed to start for some reason.
+     */
+    private static void updateLibertyServer() throws Exception {
```

Comment:
If we don't change the server.xml very much, we can simply get the configuration and request the single LDAP server from it and update the connection properties.